services/subscriptions/handers_test.go

```diff
@@ -25,3 +25,21 @@ func TestGetSubscriptionsHandler(t *testing.T) {
 		assert.Contains(t, rec.Body.String(), "{\"value\":null,\"count\":{\"type\":\"\",\"value\":0}}")
 	}
 }
+
+// TestPostSubscriptionsHandler sets up a server and tests the endpoint directly
+func TestPostSubscriptionsHandler(t *testing.T) {
+	e := echo.New()
+	req := httptest.NewRequest(http.MethodPost, "/providers/Microsoft.Billing/billingAccounts/testAccount/billingProfiles/testProfile/invoiceSections/testSection/providers/Microsoft.Subscription/createSubscription?api-version=2018-11-01-preview", nil)
+	rec := httptest.NewRecorder()
+	ctx := e.NewContext(req, rec)
+
+	subscriptionService := SubscriptionService{}
+	getHandler := subscriptionService.PostSubscriptionsHandler()
+
+	// Assertions
+	if assert.NoError(t, getHandler(ctx)) {
+		assert.Equal(t, http.StatusCreated, rec.Code)
+		// TODO: Maybe test that a UUID is returned
+		assert.Contains(t, rec.Body.String(), "{\"subscriptionLink\":\"/subscriptions/")
+	}
+}
```

Comment:
Why do we create the echo serer here, but not construct its handlers? I feel like I'm missing something but I don't get how this works.

Reply:
Looks like we need `echo.New()` for some sort of context setup since we pass the request and recorder to the Echo context before directly calling the one handler we need. I copypasta'd this code so I can't speak with 100% certainty, but my guess is that it's trying to stay as lightweight as possible. We *could* instantiate all the handlers the way we do in `main`, but that would make each test a little heavier in terms of dependencies.